File: python/tutorial/caylent_stack.py
from aws_cdk import (
    # Duration,
    Stack,
    aws_kms as kms,
    aws_s3 as s3,
    aws_sns as sns,
    aws_sqs as sqs,

    aws_s3_notifications as notifications,
    aws_sns_subscriptions as subscriptions
)
from constructs import Construct


class CaylentStack(Stack):
    def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
        super().__init__(scope, construct_id, **kwargs)

        key = kms.Key(self, "EncryptionKey")

        bucket = s3.Bucket(self, "SouceBucket", encryption_key=key)
        topic = sns.Topic(self, "EventTopic", master_key=key)
        queue1 = sqs.Queue(self, "QueueOne", encryption_master_key=key)
        queue2 = sqs.Queue(self, "QueueTwo", encryption_master_key=key)

        bucket.add_event_notifications(s3.EventType.OBJECT_CREATED_PUT, notifications.SnsDestination(topic))
        topic.add_subscription(subscriptions.SqsSubscription(queue1))
        topic.add_subscription(subscriptions.SqsSubscription(queue2))

File: typescript/lib/caylent-stack.ts
import * as cdk from "aws-cdk-lib";
import { Construct } from "constructs";
import * as kms from "aws-cdk-lib/aws-kms";
import * as s3 from "aws-cdk-lib/aws-s3";
import * as notifications from "aws-cdk-lib/aws-s3-notifications";
import * as sns from "aws-cdk-lib/aws-sns";
import * as subscriptions from 'aws-cdk-lib/aws-sns-subscriptions';
import * as sqs from "aws-cdk-lib/aws-sqs";

export class CaylentStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    const encryptionKey = new kms.Key(this, "EncryptionKey");

    const bucket = new s3.Bucket(this, "SourceBucket", {
      encryptionKey,
    });
    const topic = new sns.Topic(this, "EventTopic", {
      masterKey: encryptionKey,
    });
    const queue1 = new sqs.Queue(this, "QueueOne", {
      encryptionMasterKey: encryptionKey,
    });
    const queue2 = new sqs.Queue(this, "QueueTwo", {
      encryptionMasterKey: encryptionKey,
    });

    bucket.addEventNotification(s3.EventType.OBJECT_CREATED_PUT, new notifications.SnsDestination(topic));
    topic.addSubscription(new subscriptions.SqsSubscription(queue1))
    topic.addSubscription(new subscriptions.SqsSubscription(queue2))
  }
}
